cudamapper/src/main.cu

```diff
@@ -177,38 +182,54 @@ int main(int argc, char* argv[])
         query_target_ranges.push_back(q);
     }
 
+    // This is host cache, if it has the index it will copy it to device, if not it will generate on device and add it to host cache
+    std::map<std::pair<uint64_t, uint64_t>, std::shared_ptr<claragenomics::cudamapper::IndexCache>> host_index_cache;
+
     // This is a per-device cache, if it has the index it will return it, if not it will generate it, store and return it.
-    std::vector<std::map<std::pair<uint64_t, uint64_t>, std::shared_ptr<claragenomics::cudamapper::Index>>> index_cache(num_devices);
-
-    auto get_index = [&index_cache, max_index_cache_size](std::shared_ptr<claragenomics::DeviceAllocator> allocator,
-                                                          claragenomics::io::FastaParser& parser,
-                                                          const claragenomics::cudamapper::read_id_t start_index,
-                                                          const claragenomics::cudamapper::read_id_t end_index,
-                                                          const std::uint64_t k,
-                                                          const std::uint64_t w,
-                                                          const int device_id,
-                                                          const bool allow_cache_index) {
+    std::vector<std::map<std::pair<uint64_t, uint64_t>, std::shared_ptr<claragenomics::cudamapper::Index>>> device_index_cache(num_devices);
+
+    auto get_index = [&device_index_cache, &host_index_cache, max_index_cache_size_on_device, max_index_cache_size_on_host](std::shared_ptr<claragenomics::DeviceAllocator> allocator,
+                                                                                                                            claragenomics::io::FastaParser& parser,
+                                                                                                                            const claragenomics::cudamapper::read_id_t start_index,
```

Comment:
What happened to formatting here? (might be something silly `make format` did)

Reply:
yes, make format pushed everything to the right. will change and see if check-format complains about it.

Reply:
Ok. There are a few cases where `make format` does strange things, especially with lambdas. If it insists on having certain formatting just let it change whatever is needed

Reply:
Yes, tried again, but check-format didn't like it and make format address those spaces again.